ip_access_rules.go

```diff
@@ -0,0 +1,97 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/goccy/go-json"
+)
+
+type ListIpAccessRulesOrderOption string
+type ListIpAccessRulesMatchOption string
+type RulesModeOption string
```

Comment:
we need to prefix this to prevent it polluting the global namespace

```suggestion
type IPAccessRulesModeOption string
```

Reply:
Yea, I'm surprised there isn't a package for each resource to avoid conflicts and make things easier to find.

Kind of tough when there are 25+ `CreateX()` functions in the main package, no?

Maybe for v2 it could be `cloudflare.zone.Operation()` like the Python library?

Reply:
this is already in the works; see https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md for details.

Reply:
You may consider using typed functional options in addition to defining specific request structs in the new version so that if more options are created or needed across the library, you don't have to make breaking changes.

here's an example where I added the missing `per_page` option for custom hostnames in our version of this library (breaking change):

https://github.com/onrocketdotcom/cloudflare-go/pull/3/files#diff-432b285f81cb984bfd1ce00fa760be757c96784135a1b9ecd49d3b6c9a26522bR508

For example, this example in experimental.md would change from:

```
dParams := &cloudflare.DNSRecordListParams{
  AccountID: "d8e8fca2dc0f896fd7cb4cb0031ba249"
}
r, _, _ := c.DNSRecord.List(context.TODO(), dParams)
```

to

```
dParams := &cloudflare.DNSRecordListParams{
  AccountID: "d8e8fca2dc0f896fd7cb4cb0031ba249"
}
r, _, _ := c.DNSRecord.List(context.TODO(), dParams, WithPaginationOptions(), WithFilterOptions())
```

I don't see function definitions listed in expiremental.md so perhaps that's already what you're thinking.